File: python/04_policy_block.py
"""Handle a policy block, then retry after a redact-mode profile in the portal.

    export TONIA_API_KEY=tonia_sk_…
    python 04_policy_block.py
"""

from __future__ import annotations

import os
import sys

from tonia import PolicyBlockError, Tonia

with Tonia(api_key=os.environ["TONIA_API_KEY"]) as client:
    listed = client.models.list()
    ids = [model["id"] for model in listed["data"]]
    if not ids:
        raise SystemExit("empty allowlist — do not guess a model id")
    try:
        client.chat.completions.create(
            model=ids[0],
            messages=[{"role": "user", "content": "…"}],
        )
    except PolicyBlockError as err:
        print("Request blocked by Workspace Policies.", file=sys.stderr)
        print(
            "Ask a Workspace admin to bind this key to a redact-mode profile "
            "(Policies → Profiles), then retry.",
            file=sys.stderr,
        )
        print(err.policy_block, file=sys.stderr)
        raise SystemExit(1) from err

File: typescript/04-policy-block.ts
/**
 * Handle a policy block, then retry after a redact-mode profile is configured
 * in the tonia portal (https://portal.tonia.ca) under Policies → Profiles.
 *
 *   export TONIA_API_KEY=tonia_sk_…
 *   npx tsx 04-policy-block.ts
 */
import { PolicyBlockError, Tonia } from "@tonia/sdk";

const client = new Tonia({ apiKey: process.env.TONIA_API_KEY });

const { data } = await client.models.list();
if (!data[0]) {
  throw new Error("empty allowlist — do not guess a model id");
}

try {
  await client.chat.completions.create({
    model: data[0].id,
    messages: [{ role: "user", content: "…" }],
  });
} catch (err) {
  if (err instanceof PolicyBlockError) {
    console.error("Request blocked by Workspace Policies.");
    console.error(
      "Ask a Workspace admin to bind this key to a redact-mode profile (Policies → Profiles), then retry.",
    );
    console.error(err.policyBlock);
    process.exitCode = 1;
  } else {
    throw err;
  }
}
